src/layer/Label.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * L.Label is used for displaying small texts on the map.
+ */
+
+L.Label = L.PopupBase.extend({
+
+	options: {
+		pane: 'labelPane',
+		offset: [12, -15],
+		direction: 'right',
+		static: false,  // Reserved word, use "permanent" instead?
+		followMouse: false,
+		clickable: false,
+		// className: '',
+		zoomAnimation: true
+	},
+
+	openOn: function (map) {
```

Comment:
A big part of this still duplicated code between `L.Label` and `L.Popup` comes from the need of having `this._label` or `this._popup` on the source. That's important because one can have both a label and a popup and we don't want that opening one would close the other.

Nevertheless this code duplication is under-optimal. It's theoretically possible to factorize and have a sort dynamic property naming (`this[prop]`) but at this stage I don't see any elegant (simple, readable and maintainable) way to achieve this. Any suggestion here is welcome :)